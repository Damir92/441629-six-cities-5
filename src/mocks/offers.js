export const offers = [
  {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!',
    'goods': [
      'Fridge',
      'Washer',
      'Coffee machine',
      'Washing machine',
      'Dishwasher',
      'Laptop friendly workspace',
      'Cable TV',
      'Air conditioning',
      'Baby seat',
      'Towels',
      'Breakfast',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 2,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.37454,
      'longitude': 4.881976,
      'zoom': 16,
    },
    'volume': 1,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
    'price': 111,
    'rating': 3.3,
    'title': 'Wood and stone place',
    'type': 'room',
  },
  {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.',
    'goods': [ 'Laptop friendly workspace', 'Baby seat', 'Towels', 'Air conditioning', 'Breakfast', 'Washer' ],
    'host': { 'avatar': 'img/avatar-angelina.jpg', 'id': 25, 'isPro': true, 'name': 'Angelina' },
    'id': 4,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': { 'latitude': 52.35054, 'longitude': 4.908976, 'zoom': 16 },
    'volume': 2,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
    'price': 294,
    'rating': 3.8,
    'title': 'Loft Studio in the Central Area',
    'type': 'hotel',
  }, {
    'bedrooms': 4,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    'goods': [
      'Air conditioning',
      'Baby seat',
      'Fridge',
      'Towels',
      'Breakfast',
      'Washer',
      'Laptop friendly workspace',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 9,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.37854,
      'longitude': 4.894976,
      'zoom': 16,
    },
    'volume': 10,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
    'price': 406,
    'rating': 2.2,
    'title': 'Loft Studio in the Central Area',
    'type': 'hotel',
  }, {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.',
    'goods': [
      'Washer',
      'Towels',
      'Breakfast',
      'Air conditioning',
      'Baby seat',
      'Fridge',
      'Laptop friendly workspace',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 19,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.397540000000006,
      'longitude': 4.9099759999999995,
      'zoom': 16,
    },
    'volume': 2,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
    'price': 286,
    'rating': 2.8,
    'title': 'The Joshua Tree House',
    'type': 'room',
  }, {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    'goods': [
      'Coffee machine',
      'Breakfast',
      'Laptop friendly workspace',
      'Air conditioning',
      'Towels',
      'Baby seat',
      'Dishwasher',
      'Washer',
      'Fridge',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 22,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
    ],
    'isFavorite': false,
    'isPremium': true,
    'location': {
      'latitude': 52.37554,
      'longitude': 4.9019759999999994,
      'zoom': 16,
    },
    'volume': 5,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    'price': 143,
    'rating': 4.6,
    'title': 'Canal View Prinsengracht',
    'type': 'house',
  }, {
    'bedrooms': 2,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.',
    'goods': [
      'Washer',
      'Fridge',
      'Air conditioning',
      'Towels',
      'Baby seat',
      'Breakfast',
      'Laptop friendly workspace',
      'Dishwasher',
      'Coffee machine',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25, 'isPro': true,
      'name': 'Angelina',
    },
    'id': 24,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.367540000000005,
      'longitude': 4.883976,
      'zoom': 16,
    },
    'volume': 9,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    'price': 289,
    'rating': 2,
    'title': 'House in countryside',
    'type': 'house',
  },
];
