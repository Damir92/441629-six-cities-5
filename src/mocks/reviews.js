export const reviews = [
  {
    id: 1,
    avatar: `img/avatar-max.jpg`,
    date: `2019-04-24`,
    dateHuman: `April 2019`,
    name: `Max`,
    rating: 4,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
  {
    id: 2,
    avatar: `img/avatar-max.jpg`,
    date: `2019-04-24`,
    dateHuman: `April 2019`,
    name: `Max`,
    rating: 4,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  },
];
