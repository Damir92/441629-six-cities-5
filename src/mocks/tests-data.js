export const cityOffers = [
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    goods: [
      `Fridge`,
      `Washer`,
      `Coffee machine`,
      `Washing machine`,
      `Dishwasher`,
      `Laptop friendly workspace`,
      `Cable TV`,
      `Air conditioning`,
      `Baby seat`,
      `Towels`,
      `Breakfast`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37454,
      longitude: 4.881976,
      zoom: 16,
    },
    volume: 1,
    preview: `previewURL`,
    price: 111,
    rating: 3.3,
    title: `Wood and stone place`,
    type: `room`,
  },
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
    goods: [
      `Laptop friendly workspace`,
      `Baby seat`,
      `Towels`,
      `Air conditioning`,
      `Breakfast`,
      `Washer`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 4,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: true,
    isPremium: false,
    location: {
      latitude: 52.35054,
      longitude: 4.908976,
      zoom: 16,
    },
    volume: 2,
    preview: `previewURL`,
    price: 294,
    rating: 3.8,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  }, {
    bedrooms: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a "slow" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    goods: [
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Towels`,
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 9,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37854,
      longitude: 4.894976,
      zoom: 16,
    },
    volume: 10,
    preview: `previewURL`,
    price: 406,
    rating: 2.2,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  },
];
