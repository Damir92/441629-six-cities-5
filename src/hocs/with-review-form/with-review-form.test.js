import React, {Fragment} from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';

import {noop} from '../../mocks/tests-data';

import withReviewForm from './with-review-form';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <Fragment>
      {children}
    </Fragment>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withReviewForm(MockComponent);

describe(`with-review-form hoc rendered correctly`, () => {
  it(`withReviewForm without additional props rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        id={0}
        onSubmit={noop}
      >
        <Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`withReviewForm with all props rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        id={0}
        onSubmit={noop}
        rating={3}
        review={'test'}
      >
        <Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
