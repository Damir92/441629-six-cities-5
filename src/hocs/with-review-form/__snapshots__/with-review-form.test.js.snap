// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-auth-form hoc rendered correctly withReviewForm with all props rendered correctly 1`] = `null`;

exports[`with-auth-form hoc rendered correctly withReviewForm without additional props rendered correctly 1`] = `null`;
